bindings/pyroot/pythonizations/test/uhi_test_suite.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import numpy as np
+import pytest
+import ROOT
+import uhi.testing.indexing
+from ROOT._pythonization._uhi import _temporarily_disable_add_directory
+
+
+def th1_eq(self, other):
+    if self is other:
+        return True
+    return (
+        isinstance(other, type(self))
+        and self.kind == other.kind
+        and np.array_equal(self.values(), other.values())
+        and np.array_equal(self.counts(), other.counts())
+        and np.array_equal(self.variances(), other.variances())
+        and all(a == b for a, b in zip(self.axes, other.axes))
+    )
+
+
+@pytest.fixture(autouse=True)
+def patch_th1_eq(monkeypatch):
+    """
+    Temporarily patch ROOT.TH1.__eq__ for tests
+    Since __eq__ pythonizations are not defined, the comparison
+    would fall back to object identity (`is`) and would fail
+    """
+    monkeypatch.setattr(ROOT.TH1, "__eq__", th1_eq)
+
+
+class TestIndexing1D(uhi.testing.indexing.Indexing1D[ROOT.TH1]):
```

Comment:
Minor thing: It looks weird to have `ROOT.TH1` as the input class for all three dimensions tests. I wonder if it would still work if we had `ROOT.TH2` for the Indexing2D and `ROOT.TH3` for the Indexing3D?